components/builder-jobsrv/src/server/handlers.rs

```diff
@@ -258,29 +248,27 @@ pub fn job_group_cancel(
     RpcMessage::make(&net::NetOk::new()).map_err(Error::BuilderCore)
 }
 
-fn is_project_buildable(conn: &mut RouteClient, project_name: &str) -> bool {
-    let mut project_get = originsrv::OriginProjectGet::new();
-    project_get.set_name(String::from(project_name));
+fn is_project_buildable(state: &AppState, project_name: &str) -> bool {
+    let conn = match state.db.get_conn().map_err(Error::Db) {
+        Ok(conn_ref) => conn_ref,
+        Err(_) => return false,
+    };
 
-    match conn.route::<originsrv::OriginProjectGet, originsrv::OriginProject>(&project_get) {
-        Ok(project) => project.get_auto_build(),
+    match Project::get(project_name.to_owned(), &*conn) {
```

Comment:
At first I was against this pattern of not having the args to the model be in a struct, but I've run across a couple of instances like this where pragmatism definitely wins out